devices/IAnalogSensorToIWear/src/IAnalogSensorToIWear.cpp

```diff
@@ -128,35 +165,36 @@ static class IAnalogSensorHandler
         return true;
     }
 
-} handler;
+};
 
 // ================================
 // WEARABLE SENSORS IMPLEMENTATIONS
 // ================================
 
-class ForceTorque6DSensor : public IForceTorque6DSensor
+class IAnalogSensorToIWear::Impl::ForceTorque6DSensor : public IForceTorque6DSensor
 {
 public:
     unsigned offset = 0;
     bool groundReactionFT;
+    std::shared_ptr<IAnalogSensorHandler> handler;
 
-    ForceTorque6DSensor(SensorName name, SensorStatus status = SensorStatus::Unknown)
-        : IForceTorque6DSensor(name, status)
+    ForceTorque6DSensor(SensorName name, SensorStatus status = SensorStatus::Unknown, std::shared_ptr<IAnalogSensorHandler> analogSensorHandler = nullptr)
+        : IForceTorque6DSensor(name, status), handler(analogSensorHandler)
     {}
 
     void setStatus(const SensorStatus status) { m_status = status; }
 
     bool getForceTorque6D(Vector3& force3D, Vector3& torque3D) const override
     {
-        bool dataOk = handler.readData();
+        bool dataOk = handler->readData();
 
         // Dirty workaround to set the status from a const method
         auto nonConstThis = const_cast<ForceTorque6DSensor*>(this);
-        nonConstThis->setStatus(handler.getStatus());
+        nonConstThis->setStatus(handler->getStatus());
 
         // TODO: The positions of force and torques are hardcoded. Forces should be the first
         //       triplet of elements of the read vector and torques the second one.
-        bool ok = dataOk && handler.getData(force3D, offset) && handler.getData(torque3D, offset + 3);
+        bool ok = dataOk && handler->getData(force3D, offset) && handler->getData(torque3D, offset + 3);
         if (groundReactionFT) {
             force3D[0] = -1 * force3D[0];
```

Comment:
Btw I find these `-1` very sketchy and confusing. I have an idea about why this is required but I don't think this is the right place to do it.

Since there's this open PR on this device, I would either add a comment explaining the intuition behind the existence of these `-1`, or move them downstream code (or, in the attached `IAnalogSensor` interface).

This device is generic and these constants do not have to be applied in any general case.

Reply:
> This device is generic and these constants do not have to be applied in any general case.

It is not applied for any general case, but this transform is triggered by flag from the configuration file. However, I agree that probably we may find a more appropriate location for applying it. I have opened a specific issue to discuss this https://github.com/robotology/wearables/issues/53.